src/lib/game.ts

```diff
@@ -87,14 +87,16 @@ export function generateEmojiArt(
   const medals = guessModeMedal(guessMode);
   const count = gameLost(solution, rows) ? "X" : rows.length;
   let result = `${name}${medals} ${gameDay + 1} ${count}/${ROW_COUNT}\n`;
-  for (let i = 0; i < rows.length; ++i) {
-    const row = rows[i];
-    for (const x of findRowStates(solution, row)) {
+  for (const [i, row] of rows.entries()) {
+    const rowStates = findRowStates(solution, row);
+    for (const x of rowStates) {
       result += x === State.Correct ? "🟩" : x === State.Present ? "🟨" : "⬛";
     }
-    if (i < rows.length - 1) {
+    
+    // Only include Discord spoilers and new lines if we haven't won yet
+    if (!rowStates.every((s) => s === State.Correct)) {
       if (discord) result += ` ||\`${row.toUpperCase()}\`||`;
-      result += "\n";
+      if (i < ROW_COUNT - 1) result += "\n";
```

Comment:
This is actually independent of the spoiler functionality so I think it's a bit misleading to nest it like that

Reply:
Hmm, it's independent of the spoiler functionality, but it *is* dependent on whether or not you've won the game.

I agree it is a little confusing though. The alternative would be:

```suggestion
    // Only include Discord spoilers if we haven't won yet
    if (discord && row !== solution) result += ` ||\`${row.toUpperCase()}\`||`;
    if (i < rows.length - 1) result += "\n";
```

Since the `result += "\n"` is now executed for every solution, we can't use `ROW_LENGTH`.